src/ol/proj/proj.js

```diff
@@ -278,6 +282,32 @@ ol.proj.Projection.prototype.setGlobal = function(global) {
 
 
 /**
+ * Does this projection have nominally constant point resolution?
+ * @return {boolean} Whether the projection has nominally constant point
+ * resolution.
+ * @api
+ */
+ol.proj.Projection.prototype.hasConstantPointResolution = function() {
+  return this.constantPointResolution_;
+};
+
+
+/**
+ * Set if the projection has nominally constant point resolution.
+ * The attribute can not be set for the EPSG:3857 projection.
+ * It should only be set for projections with nominally constant scale
+ * or those with units of Pixels or Degrees.
+ * @api
+ */
+ol.proj.Projection.prototype.setConstantPointResolution =
+    function() {
+  if (this.code_ != 'EPSG:3857') {
+    this.constantPointResolution_ = true;
+  }
```

Comment:
Is it the 3857 check that bothers you ?
Actually, as getPoint Resolution is overridden for 3857, the check for 3857 could be dropped.